apcd-cms/src/client/src/components/Forms/Registrations/RegistrationForm.tsx

```diff
@@ -0,0 +1,504 @@
+import React, { useState, useEffect } from 'react';
+import { Formik, Form, Field, ErrorMessage } from "formik";
+import * as Yup from "yup";
+import {
+  FormGroup,
+  Label,
+  Button,
+  FormFeedback,
+} from "reactstrap";
+import { useFormData } from 'hooks/registrations';
+import { fetchUtil } from 'utils/fetchUtil';
+import USStates from './USStates.fixture';
+import { TextFormField } from './TextFormField';
+import { RegistrationEntity } from './FormEntity';
+import { RegistrationContact } from './FormContact';
+import SectionMessage from 'core-components/SectionMessage';
+import LoadingSpinner from 'core-components/LoadingSpinner';
+import styles from './RegistrationForm.module.css';
+
+const validationSchema = Yup.object().shape({
+    reg_year: Yup.string().matches(/^(202[3-9]|20[3-9][0-9]|2100)$/, { message: 'Registration year must be 2023 or later' }).required('Registration year is required'),
+    business_name: Yup.string().required('Business name is required'),
+    mailing_address: Yup.string().required('Mailing address is required'),
+    city: Yup.string().required('City is required'),
+    zip_code: Yup.string()
+      .matches(/^\d{5}(-\d{4})?$/, { message: 'ZIP is not properly formatted' })
+      .required('ZIP Code is required'),
+    entities: Yup.array().of(
+        Yup.object().shape({
+          entity_name: Yup.string().required('Entity name is required'),
+          fein: Yup.string().matches(/^\d{2}-\d{7}$/, { message: 'FEIN is not properly formatted' }),
+          license_number: Yup.string().matches(/^(?!0+$)[0-9]{1,10}$/, { message: 'License no. is not properly formatted' }),
+          naic_company_code: Yup.string().matches(/^(?!0+$)[0-9]{1,10}$/, { message: 'NAIC code is not properly formatted' }),
+          types_of_plans_commercial: Yup.boolean(),
+          types_of_plans_medicare: Yup.boolean(),
+          types_of_plans_medicaid: Yup.boolean(),
+          types_of_files_eligibility_enrollment: Yup.boolean(),
+          types_of_files_provider: Yup.boolean(),
+          types_of_files_medical: Yup.boolean(),
+          types_of_files_pharmacy: Yup.boolean(),
+          types_of_files_dental: Yup.boolean(),
+          total_covered_lives: Yup.number().positive()
+            .required('Total covered lives is required'),
+          claims_encounters_volume: Yup.number().positive()
+            .required('Claims and Encounters volume is required'),
+          total_claims_value: Yup.number().positive()
```

Comment:
The error message could be a bit more user friendly when not the right type. 
![Screenshot 2024-11-01 at 9 18 37 AM](https://github.com/user-attachments/assets/e0a58604-00f3-45a7-b813-d707e46075d3)